databallpy/match.py

```diff
@@ -621,6 +623,39 @@ def get_match(
         axis=1,
     )
 
+    if (
+        not set(event_metadata.home_players["id"])
+        == set(tracking_metadata.home_players["id"])
+    ) or (
+        not set(event_metadata.away_players["id"])
+        == set(tracking_metadata.away_players["id"])
+    ):
+        full_name_id_map = {}
+        for idx, row in event_metadata.home_players.iterrows():
+            full_name_tracking_metadata = get_matching_full_name(
+                row["full_name"], tracking_metadata.home_players["full_name"]
+            )
+            id_tracking_data = tracking_metadata.home_players.loc[
+                tracking_metadata.home_players["full_name"]
+                == full_name_tracking_metadata,
+                "id",
+            ].values[0]
+            event_metadata.home_players.loc[idx, "id"] = id_tracking_data
+            full_name_id_map[row["full_name"]] = id_tracking_data
+
+        for idx, row in event_metadata.away_players.iterrows():
+            full_name_tracking_metadata = get_matching_full_name(
+                row["full_name"], tracking_metadata.away_players["full_name"]
+            )
+            id_tracking_data = tracking_metadata.away_players.loc[
+                tracking_metadata.away_players["full_name"]
+                == full_name_tracking_metadata,
+                "id",
+            ].values[0]
+            event_metadata.away_players.loc[idx, "id"] = id_tracking_data
+            full_name_id_map[row["full_name"]] = id_tracking_data
+        event_data["player_id"] = event_data["player_name"].map(full_name_id_map)
```

Comment:
Annoying that the id's are not alligned, I would suggest making a different function to keep the `get_match` function as compact as possible. Keep the if statement and create a function `align_player_ids` or something like that, that also changes the event_data like you are doing now. Make sure to add docstrings and tests for the new functions